libsolidity/interface/CompilerStack.cpp

```diff
@@ -1206,13 +1205,75 @@ Json CompilerStack::ethdebug(Contract const& _contract, bool _runtime) const
 	solAssert(m_stackState >= AnalysisSuccessful, "Analysis was not successful.");
 	solAssert(_contract.contract);
 	solUnimplementedAssert(!isExperimentalSolidity());
+	Json result = Json::object();
+	Json contract = Json::object();
+	contract["name"] = _contract.contract->name();
+	Json definition = Json::object();
+	definition["sources"]["id"] = static_cast<int>(sourceIndices()[_contract.contract->sourceUnitName()]);
+	contract["definition"] = definition;
+	result["contract"] = contract;
+	result["environment"] = _runtime ? "call" : "create";
+	result["instructions"] = ethdebugInstructions(_contract, _runtime);
+	return result;
+}
+
+Json CompilerStack::ethdebugInstructions(Contract const& _contract, bool _runtime) const
+{
+	evmasm::LinkerObject const* object;
+	evmasm::Assembly const* assembly;
 	if (_runtime)
 	{
-		Json result = Json::object();
-		return result;
+		object = &_contract.runtimeObject;
+		assembly = _contract.evmRuntimeAssembly.get();
 	}
-	Json result = Json::object();
-	return result;
+	else
+	{
+		object = &_contract.object;
+		assembly = _contract.evmAssembly.get();
+	}
+
```

Comment:
Instead of using pointers you could also use ternary like `evmasm::LinkerObject const& object = _runtime ? _contract.runtimeObject : _contract.object`